### PR TITLE
Make size a property with an assertion

### LINKED ISSUE #120568: Ban or change behavior of TensorVariable.size
### 🐛 Describe the bug

Recent bug https://github.com/pytorch/pytorch/issues/120482 was due to someone directly accessing TensorVariable.size; fixed in https://github.com/pytorch/pytorch/pull/120567. But this field is None when the tensor has dynamic shapes. This is a dynamic shapes footgun. Either ban this, or change this to query size off of the fake tensor (and be willing to return SymInts).

### Versions

main

cc @msaroufim @bdhirsh @anijain2305 @zou3519 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @aakhundov

### PR DESCRIPTION
**Pull request by mlazos**
Fixes https://github.com/pytorch/pytorch/issues/120568

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by williamwen42 — APPROVED** [review]

File: torch/_dynamo/variables/tensor.py [hunk 1]
```diff
@@ -503,6 +503,11 @@ def unpack_var_sequence(self, tx: "InstructionTranslator", idxes=None):
             for i in idxes
         ]
 
+    @property
+    def size(self):
+        assert self._size, "accessing None size in TensorVariable"
```
[the remark this answers is not in the document]
Comment (mlazos) [reply torch/_dynamo/variables/tensor.py:508]: wdym? I think this assert works as intended, we do this in a ton of places

[the remark this answers is not in the document]
Comment (mlazos) [reply torch/_dynamo/variables/tensor.py]: I updated this in case there is just a zero or something here, or an empty list

**Pull request closed without merging**